ssmconfig.go

```diff
@@ -84,9 +90,16 @@ func (p *Provider) Process(configPath string, c interface{}) error {
 			continue
 		}
 
-		err = setValue(v.Field(i), value)
+		valueField := v.Field(i)
+		typeField := t.Field(i)
+		var err error
+		if isTextUnmarshaler(typeField) {
+			err = unmarshalText(typeField, valueField, value)
+		} else {
+			err = setValue(valueField, value)
+		}
```

Comment:
I would move all of the logic for `TextUnmarshaler` into the `setValue` function. 

We also need to handle the case where the value is already a pointer to the implementation of `Text Unmarshaled`. 

Something like:
```go
func setValue(v reflect.Value, s string) error {
	t := v.Type()

	if t.Implements(textUnmarshalerType) {
		return unmarshalText(t, v, s, /*shouldAddr*/ false)
	}
	if reflect.PtrTo(t).Implements(textUnmarshalerType) {
		return unmarshalText(t, v, s, /*shouldAddr*/ true)
	}
	
	//...
}
```